web/auth/util.go

```diff
@@ -1,16 +1,37 @@
 package auth
 
+import (
+	"strings"
+)
+
 // GetCallbackURL returns the callback URL for a given base URL and a provider.
 func GetCallbackURL(baseURL, provider string) string {
-	return GetProviderURL(baseURL, "auth", provider, "callback")
+	return GetProviderURL(baseURL, "auth/callback", provider)
 }
 
 // GetEventsURL returns the event URL for a given base URL and a provider.
 func GetEventsURL(baseURL, provider string) string {
-	return GetProviderURL(baseURL, "hook", provider, "events")
+	return GetProviderURL(baseURL, "hook", provider)
 }
 
 // GetProviderURL returns a URL endpoint given a base URL and a provider.
-func GetProviderURL(baseURL, route, provider, endpoint string) string {
-	return "https://" + baseURL + "/" + route + "/" + provider + "/" + endpoint
+func GetProviderURL(baseURL, route, provider string) string {
```

Comment:
This method should be unexported; it is only used by the two methods above. Moreover, please add a TODO here and remind us to replace this with the new [`URL.JoinPath()`](https://pkg.go.dev/net/url@master#URL.JoinPath) coming in Go 1.19.